### PR TITLE
creating restTemplate object in postconstruct

### PR DESCRIPTION
**Pull request by Ajay1596**

### CODE REVIEW

File: kernel/kernel-auth-adapter/src/main/java/io/mosip/kernel/auth/defaultadapter/handler/AuthHandler.java [hunk 1]
```diff
@@ -92,11 +103,33 @@
...
 	@Autowired
 	private ObjectMapper objectMapper;
-	
-	@Autowired
-	private RestTemplate restTemplate;
-	
+
+	private RestTemplate restTemplate = null;
+
 	private static final String DEFAULTADMIN_MOSIP_IO = "defaultadmin@mosip.io";
```
Comment (gsasikumar) [inline kernel/kernel-auth-adapter/src/main/java/io/mosip/kernel/auth/defaultadapter/handler/AuthHandler.java:114]: What is this? I know u have not changed but why is it not in configuration with this as a default value? if this is not used please remove it.

File: kernel/kernel-auth-adapter/src/main/java/io/mosip/kernel/auth/defaultadapter/handler/AuthHandler.java [hunk 2]
```diff
@@ -361,9 +399,9 @@ private void sendErrors(RoutingContext routingContext, List<ServiceError> errors
 			LOGGER.error(exception.getMessage());
 		}
 	}
-	
+
 	public String getContextUser() {
-		Authentication authentication=SecurityContextHolder.getContext().getAuthentication();
-		return authentication==null?DEFAULTADMIN_MOSIP_IO:authentication.getName();
+		Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+		return authentication == null ? DEFAULTADMIN_MOSIP_IO : authentication.getName();
```
Comment (gsasikumar) [inline kernel/kernel-auth-adapter/src/main/java/io/mosip/kernel/auth/defaultadapter/handler/AuthHandler.java:413]: why should it give this user? why not throw an exception? or send empty? What's the use of this?

File: kernel/kernel-auth-adapter/src/main/java/io/mosip/kernel/auth/defaultadapter/handler/AuthHandler.java [hunk 3]
```diff
@@ -189,27 +223,31 @@ private MosipUserDto buildDto(Claims claims) {
...
+	 * HttpClients.custom().setSSLSocketFactory(csf).build(); //
+	 * HttpComponentsClientHttpRequestFactory requestFactory = new
+	 * HttpComponentsClientHttpRequestFactory(); //
+	 * requestFactory.setHttpClient(httpClient); // RestTemplate restTemplate = new
+	 * RestTemplate(requestFactory); RestTemplate restTemplate = new RestTemplate();
+	 * restTemplate.setInterceptors(Collections.singletonList(new
+	 * RestTemplateInterceptor())); return restTemplate; }
+	 */
 
 	public void addCorsFilter(HttpServer httpServer, Vertx vertx) {
```
Comment (gsasikumar) [inline kernel/kernel-auth-adapter/src/main/java/io/mosip/kernel/auth/defaultadapter/handler/AuthHandler.java:260]: what CORS is it? rename this method to make sense.

**Review by gsasikumar — CHANGES_REQUESTED** [review]

**Comment by gsasikumar** [comment]
@Ajay1596 also have you checked the expiry of token scenario?

**Comment by urviljoshi** [comment]
closed as discussed with @gsasikumar

**Pull request closed without merging**